### ISSUE TITLE
`TRY` / `FINALLY` when used with `WITH DOCKER` produces verbose output

### ISSUE THREAD

**Issue by negz** [earthly/earthly#4173]
consider:

```
VERSION --try 0.8

test:
  FROM earthly/dind:alpine
  TRY
    WITH DOCKER
      RUN echo important > data && false
    END
  FINALLY
    SAVE ARTIFACT data AS LOCAL ./data
  END
```

running `+test` will output:

```
               +test *failed* | Repeating the failure error...
               +test *failed* | --> WITH DOCKER RUN --privileged echo important > data && false
               +test *failed* | Starting dockerd with data root /var/earthly/dind/7ce618e5e3ddb8ca57e6b55c62c59b231e26d751e3736a2340e5c889f5127a58/tmp.EeNbaL
               +test *failed* | ERROR Earthfile:7:6
               +test *failed* |       The command
               +test *failed* |           WITH DOCKER RUN --privileged echo important > data && false
               +test *failed* |       did not complete successfully. Exit code 1
               +test | unlazy force execution: process "/bin/sh -c EARTHLY_DOCKERD_DATA_ROOT=\"/var/earthly/dind/7ce618e5e3ddb8ca57e6b55c62c59b231e26d751e3736a2340e5c889f5127a58\" EARTHLY_DOCKERD_CACHE_DATA=\"false\" EARTHLY_DOCKER_LOAD_FILES=\"\" EARTHLY_IMAGES_WITH_DIGESTS=\"\" EARTHLY_START_COMPOSE=\"false\" EARTHLY_COMPOSE_FILES=\"\" EARTHLY_COMPOSE_SERVICES=\"\" PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin /var/earthly/dockerd-wrapper.sh execute /usr/bin/earth_debugger /bin/sh -c 'echo important > data && false'" did not complete successfully: exit code: 1
```

(but does correctly saves the `data` file to the local directory)

where as when `TRY` / `FINALLY` without `WITH DOCKER`, e.g.

```
VERSION --try 0.8

test:
  FROM earthly/dind:alpine
  TRY
    RUN echo important > data && false
  FINALLY
    SAVE ARTIFACT data AS LOCAL ./data
  END
```

produces much easier to understand output:

```
               +test *failed* | Repeating the failure error...
               +test *failed* | --> RUN echo important > data && false
               +test *failed* | [no output]
               +test *failed* | ERROR Earthfile:6:4
               +test *failed* |       The command
               +test *failed* |           RUN echo important > data && false
               +test *failed* |       did not complete successfully. Exit code 1

```

Note that this issue originally reported a second issue here, which relates to the ordering of `WITH DOCKER` and `TRY`/`FINALLY`, which is tracked under a separate issue: #4177

----
original report below:

### Question

I notice that when this `RUN` exits non-zero the `FINALLY SAVE ARTIFACT` doesn't happen.

https://github.com/crossplane/crossplane/blob/ed0fb98cc8d19a96229caf460f2b0950d583438f/Earthfile#L60

I also see the following:

```
unlazy force execution: process "/bin/sh -c EARTHLY_DOCKERD_DATA_ROOT=\"/var/earthly/dind/944cc3b666f2b62bc87a5159517f6f8d1619489218402a5931707a886c2d7dea\" EARTHLY_DOCKERD_CACHE_DATA=\"false\" EARTHLY_DOCKER_LOAD_FILES=\"\" EARTHLY_IMAGES_WITH_DIGESTS=\"crossplane-e2e/crossplane:latest@sha256:2427e9fcd06f0f5efde8c7157db825d08402e2239ce69d457272f9c96f3bd182\" EARTHLY_START_COMPOSE=\"false\" EARTHLY_COMPOSE_FILES=\"\" EARTHLY_COMPOSE_SERVICES=\"\" FLAGS='-test.failfast -fail-fast --test-suite ssa-claims' GOLANG_VERSION=1.22.3 GOPATH=/go GOTOOLCHAIN=local GO_VERSION=1.22.3 PATH=/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin /var/earthly/dockerd-wrapper.sh execute /usr/bin/earth_debugger /bin/sh -c 'gotestsum --no-color=false --format testname --junitfile e2e-tests.xml --raw-command go tool test2json -t -p E2E ./e2e -test.v ${FLAGS}'" did not complete successfully: exit code: 1
```

Is this just a symptom of a normal error? I notice that other targets seem to return non-zero without producing an `unlazy force execution` error, so I'm wondering whether it's related to my output not being saved or just a red herring.

Perhaps related to https://github.com/earthly/earthly/issues/2742.

**Comment by adamgordonbell** [comment]
> Is this just a symptom of a normal error? I notice that other targets seem to return non-zero without producing an unlazy force execution error, so I'm wondering whether it's related to my output not being saved or just a red herring.

`unlazy force execution` is most likely just a non-zero exit code. I believe its just an error message that needs to be improved. It's on my todo list.

FINALLY not running in WITH DOCKER would be a bigger issue. Let me see if I can reproduce that.

**Comment by adamgordonbell** [comment]
Tracking the error message improvements here: https://github.com/earthly/earthly/issues/4181

**Comment by adamgordonbell** [comment]
Fixed in #4180
